pkg/controllers/provisioning/provisioner.go

```diff
@@ -133,23 +133,35 @@ func (p *Provisioner) Provision(ctx context.Context) error {
 		return nil
 	}
 
+	_, err = p.LaunchNodes(ctx, LaunchOptions{RecordPodNomination: true}, nodes...)
+	return err
+}
+
+type LaunchOptions struct {
+	// RecordPodNomination causes nominate pod events to be recorded against the node.
+	RecordPodNomination bool
```

Comment:
Thoughts on checking `pod.spec.nodeName` instead of wiring through this struct.

Reply:
I prefer it explicit.